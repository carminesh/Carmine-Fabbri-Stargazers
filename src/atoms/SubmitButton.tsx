import { Image, ImageSourcePropType, ScaledSize, StyleSheet, Text, TouchableOpacity, useWindowDimensions, View } from 'react-native';
import React from 'react';
import ColorsPalette from '../constants/ColorsPalette';

interface SubmitButtonProps {
    icon?: ImageSourcePropType;
    isButtonEnabled: boolean;
    onPressCallback: () => Promise<void>;
}

const SubmitButton: React.FC<SubmitButtonProps> = ({ icon, isButtonEnabled, onPressCallback }) => {
    const dimensions = useWindowDimensions();
    const styles = style(dimensions);

    return (
        <TouchableOpacity style={[styles.buttonContainer, isButtonEnabled && { opacity: 0.7 }]} disabled={isButtonEnabled} onPress={() => onPressCallback()}>
            {icon && <Image style={styles.icon} source={icon} resizeMode="contain" />}
            <Text style={styles.buttonText}>Search</Text>
        </TouchableOpacity>
    );
};

export default SubmitButton;

const style = (dimensions: ScaledSize) =>
    StyleSheet.create({
        buttonContainer: {
            width: dimensions.width - 32,
            height: 45,
            backgroundColor: ColorsPalette.BLUE_ACCENT,
            justifyContent: 'center',
            alignItems: 'center',
            borderRadius: 5,
            flexDirection: 'row',
        },
        buttonText: {
            color: ColorsPalette.WHITE,
            fontWeight: '600',
        },
        icon: {
            tintColor: ColorsPalette.WHITE,
            marginRight: 10,
            width: 15,
            height: 15,
        },
    });
