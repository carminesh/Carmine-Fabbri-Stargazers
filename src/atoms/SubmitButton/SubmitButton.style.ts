import { ScaledSize, StyleSheet } from 'react-native';
import ColorsPalette from '../../constants/ColorsPalette';

export const style = (dimensions: ScaledSize) =>
    StyleSheet.create({
        buttonContainer: {
            width: dimensions.width - 32,
            height: 45,
            backgroundColor: ColorsPalette.BLUE_ACCENT,
            justifyContent: 'center',
            alignItems: 'center',
            borderRadius: 5,
            flexDirection: 'row',
        },
        buttonText: {
            color: ColorsPalette.WHITE,
            fontWeight: '600',
        },
        icon: {
            tintColor: ColorsPalette.WHITE,
            marginRight: 10,
            width: 15,
            height: 15,
        },
    });
