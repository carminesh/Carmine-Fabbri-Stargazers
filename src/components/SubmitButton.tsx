import { ScaledSize, StyleSheet, Text, TouchableOpacity, useWindowDimensions, View } from 'react-native';
import React from 'react';
import ColorsPalette from '../constants/ColorsPalette';

interface SubmitButtonProps {
    isButtonDisabled: boolean;
}

const SubmitButton: React.FC<SubmitButtonProps> = ({ isButtonDisabled }) => {
    const dimensions = useWindowDimensions();
    const styles = style(dimensions);

    return (
        <TouchableOpacity style={styles.buttonContainer} disabled={isButtonDisabled}>
            <Text style={styles.buttonText}>Search</Text>
        </TouchableOpacity>
    );
};

export default SubmitButton;

const style = (dimensions: ScaledSize) =>
    StyleSheet.create({
        buttonContainer: {
            width: dimensions.width - 32,
            height: 45,
            backgroundColor: ColorsPalette.BLUE_ACCENT,
            justifyContent: 'center',
            alignItems: 'center',
            borderRadius: 5,
        },
        buttonText: {
            color: ColorsPalette.WHITE,
            fontWeight: 'bold',
        },
    });
