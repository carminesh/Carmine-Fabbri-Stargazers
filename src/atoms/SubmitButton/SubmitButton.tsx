import { Image, ImageSourcePropType, Text, TouchableOpacity, useWindowDimensions } from 'react-native';
import React from 'react';
import { style } from './SubmitButton.style';

interface SubmitButtonProps {
    icon?: ImageSourcePropType;
    isButtonEnabled: boolean;
    onPressCallback: () => Promise<void>;
}

const SubmitButton: React.FC<SubmitButtonProps> = ({ icon, isButtonEnabled, onPressCallback }) => {
    const dimensions = useWindowDimensions();
    const styles = style(dimensions);

    return (
        <TouchableOpacity style={[styles.buttonContainer, isButtonEnabled && { opacity: 0.7 }]} disabled={isButtonEnabled} onPress={() => onPressCallback()}>
            {icon && <Image style={styles.icon} source={icon} resizeMode="contain" />}
            <Text style={styles.buttonText}>Search</Text>
        </TouchableOpacity>
    );
};

export default SubmitButton;

