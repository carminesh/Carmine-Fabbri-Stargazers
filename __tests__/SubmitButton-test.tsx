import 'react-native';
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import SubmitButton from '../src/atoms/SubmitButton/SubmitButton';
import renderer from 'react-test-renderer';

describe('SubmitButton testing', () => {
    it('renders correctly', () => {
        const onPressCallback = jest.fn();
        const tree = renderer.create(<SubmitButton isButtonEnabled={true} onPressCallback={onPressCallback} icon={{ uri: 'https://example.com/icon.png' }} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should call the onPressCallback function when pressed', async () => {
        const onPressCallback = jest.fn();
        const { getByTestId } = render(<SubmitButton isButtonEnabled={true} onPressCallback={onPressCallback} />);

        const button = getByTestId('submit-button');
        fireEvent.press(button);
        expect(onPressCallback).toHaveBeenCalled();
    });

    it('should not call the onPressCallback function when button is disabled', async () => {
        const onPressCallback = jest.fn();
        const { getByTestId } = render(<SubmitButton isButtonEnabled={false} onPressCallback={onPressCallback} />);

        const button = getByTestId('submit-button');
        fireEvent.press(button);
        expect(onPressCallback).not.toHaveBeenCalled();
    });

    it('should display the icon passed as a prop', async () => {
        const onPressCallback = jest.fn();
        const icon = { uri: 'https://example.com/icon.png' };
        const { queryByTestId } = render(<SubmitButton isButtonEnabled={true} onPressCallback={onPressCallback} icon={icon} />);

        const iconImage = queryByTestId('submit-button-icon');
        expect(iconImage).toBeTruthy();
    });

    it('should not display an icon if not passed as a prop', async () => {
        const onPressCallback = jest.fn();
        const { queryByTestId } = render(<SubmitButton isButtonEnabled={true} onPressCallback={onPressCallback} />);

        const iconImage = queryByTestId('submit-button-icon');
        expect(iconImage).toBeFalsy();
    });
});
