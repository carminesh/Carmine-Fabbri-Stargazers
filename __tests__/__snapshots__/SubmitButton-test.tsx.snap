// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton testing renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#2C68F7",
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "center",
      "opacity": 0.7,
      "width": 718,
    }
  }
  testID="submit-button"
>
  <Image
    resizeMode="contain"
    source={
      {
        "uri": "https://example.com/icon.png",
      }
    }
    style={
      {
        "height": 15,
        "marginRight": 10,
        "tintColor": "#FFFFFF",
        "width": 15,
      }
    }
    testID="submit-button-icon"
  />
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontWeight": "600",
      }
    }
  >
    Search
  </Text>
</View>
`;
